### PR TITLE
[fix-12687]when timeout_notify_strategy is null, modify timeout_notify_strategy

### LINKED ISSUE #12687: [Bug] [dolphinscheduler-api] in the task define,click edit,timeoutStategy status consistent
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

this is null, but not null
image

### What you expected to happen

same

### How to reproduce

1. create task
2. in the task define, click edit

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jackfanwan**
when timeout_notify_strategy is null, modify timeout_notify_strategy is 1000, prevention maps to valid values

close #12687

### CODE REVIEW

**Comment by caishunfeng** [comment]
```
public enum TaskTimeoutStrategy {

    /**
     * 0 warn
     * 1 failed
     * 2 warn+failed
     */
    WARN(0, "warn"),
    FAILED(1, "failed"),
    WARNFAILED(2, "warnfailed");

}
```
Because the default is WARN, maybe it's not a bug.

**Comment by jackfanwan** [comment]
> ```
> 
> public enum TaskTimeoutStrategy {
> 
> 
> 
>     /**
> 
>      * 0 warn
> 
>      * 1 failed
> 
>      * 2 warn+failed
> 
>      */
> 
>     WARN(0, "warn"),
> 
>     FAILED(1, "failed"),
> 
>     WARNFAILED(2, "warnfailed");
> 
> 
> 
> }
> 
> ```
> 
> Because the default is WARN, maybe it's not a bug.

because this, when edit task, equals  is false

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java [hunk 1]
```diff
@@ -716,8 +717,12 @@ private TaskDefinitionLog updateTask(User loginUser, long projectCode, long task
         }
         TaskDefinitionLog taskDefinitionToUpdate =
                 JSONUtils.parseObject(taskDefinitionJsonObj, TaskDefinitionLog.class);
+        if (TimeoutFlag.CLOSE == taskDefinition.getTimeoutFlag()) {
+            taskDefinition.setTimeoutNotifyStrategy(null);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java:721]: This field is enum type, will it error if null? Please take a check.

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**